day5/pathsum.py

```diff
@@ -0,0 +1,24 @@
+# Definition for a binary tree node.
+# class TreeNode(object):
+#     def __init__(self, x):
+#         self.val = x
+#         self.left = None
+#         self.right = None
+
+class Solution(object):
+    def hasPathSum(self, root, sum):
+        """
+        :type root: TreeNode
+        :type sum: int
+        :rtype: bool
+        """
+        # empty tree
+        if not root:
```

Comment:
prefer explicit checks against None instead of checking falsy-ness in case root has some zero case that would make `__nonzero__` (py 2) or `__bool__` (py 3) return false